### PR TITLE
142. Linked List Cycle II cpp solution

### LINKED ISSUE #47: Add LeetCode - 142. Linked List Cycle II in C++
I want to provide solution to the problem [Linked List Cycle II](https://leetcode.com/problems/linked-list-cycle-ii/) in C++.

### PR DESCRIPTION
**Pull request by NafiAsib**
- [x] Added cpp solution for the problem
- [x] Added video resources link

Resolves #47

### CODE REVIEW

File: Algorithms/Medium/142_LinkedListCycleII/Solution.cpp [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+class Solution
+{
+public:
+    ListNode *detectCycle(ListNode *head)
+    {
+        if (!head)
+        {
+            printf("no cycle\n");
+            return NULL;
+        }
```
Comment (Tahanima) [inline Algorithms/Medium/142_LinkedListCycleII/Solution.cpp:18]: Add a blank line to separate the **if** block from the initialization statement.

File: Algorithms/Medium/142_LinkedListCycleII/Solution.cpp [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        ListNode *tortoise = head, *hare = head;
+
+        while (1)
+        {
+            if (!hare->next || !(hare->next)->next)
+                break; // no cycle
+            // move hare pointer 2 times and tortoise pointer 1 times
+            tortoise = tortoise->next;
+            hare = (hare->next)->next;
```
Comment (Tahanima) [inline Algorithms/Medium/142_LinkedListCycleII/Solution.cpp:27]: Add a blank line to separate here as well.

File: Algorithms/Medium/142_LinkedListCycleII/Solution.cpp [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            if (hare == tortoise)
+            {
+                ListNode *T1 = head, *T2 = tortoise;
```
Comment (Tahanima) [inline Algorithms/Medium/142_LinkedListCycleII/Solution.cpp:30]: Give the variables **T1** and **T2** more meaningful names. Also, the variable naming convention should be consistent. So, stick to camelCase.

File: Algorithms/Medium/142_LinkedListCycleII/Solution.cpp [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+                int position = 0;
+                while (1)
+                {
+                    if (T1 == T2)
+                    {
+                        printf("tail connects to node index %d\n", position);
+                        return T1;
+                    }
```
Comment (Tahanima) [inline Algorithms/Medium/142_LinkedListCycleII/Solution.cpp:38]: Add a blank line here as well.

File: Algorithms/Medium/142_LinkedListCycleII/Solution.cpp [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        {
+            printf("no cycle\n");
+            return NULL;
+        }
+        ListNode *tortoise = head, *hare = head;
+
+        while (1)
+        {
+            if (!hare->next || !(hare->next)->next)
+                break; // no cycle
```
Comment (Tahanima) [inline Algorithms/Medium/142_LinkedListCycleII/Solution.cpp:24]: Add a blank line.

**Review by Tahanima — CHANGES_REQUESTED** [review]
Great effort and your solution looks good! Just resolve the requested changes, then your pr will be ready for merge. 

Cheers 🎉

**Comment by NafiAsib** [comment]
@Tahanima please review, changes were made according to your response

**Review by Tahanima — APPROVED** [review]
Excellent work! :fire: Thank you so much for your contribution! :grinning: and welcome to **LeetCode Solution Curation** as a **Contributor**

Cheers! 🙌

**Pull request merged**